### ISSUE TITLE
Support for importing objects as custom environments

### ISSUE THREAD

**Issue by JieamI** [vitest-dev/vitest#2904]
### Clear and concise description of the problem

> Starting from 0.23.0, you can create your own package to extend Vitest environment. To do so, create package with the name vitest-environment-${name}. That package should export an object with the shape of Environment:

Sometimes, custom environment is just a module file, I don't want to make it a npm package. Importing this module directly is a better option for me.

### Suggested solution

Allows developers to use Environment type objects directly as custom environments.
```
// test/env/custom-env.ts
import type { Environment } from 'vitest'
export default <Environment>{
  name: 'custom',
  setup() {
    ...
    return {
      teardown() {
        ...
      }
    }
  }
}


// vitest.config.ts
import customEnv from './test/env/custom-env'
export default defineConfig({
  test: {
    environment: customEnv
  }
})
```

### Validations

- [x] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [x] Read the [docs](https://vitest.dev/guide/).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by sheremet-va** [comment]
Importing objects would not be possible. But we can allow importing any file.

**Comment by JieamI** [comment]
> But we can allow specifying a path to any file.

This is also a very good idea, and I would appreciate it if this feature could be supported in future version.
> Importing objects would not be possible.

why? Is this technically impossible？

**Comment by sheremet-va** [comment]
> why? Is this technically impossible？

Yes, config is resolved in the main thread, but tests are running in worker threads, so we cannot passdown functions between them. And we don’t want to resolve config for each test because it’s expensive.

**Comment by kouak** [comment]
> Importing objects would not be possible. But we can allow specifying a path to any file.

Relative path environment implemented via #3203 and released in [v0.34.0](https://github.com/vitest-dev/vitest/releases/tag/v0.34.0)
